ouroboros-network/src/Ouroboros/Network/BlockFetch/ClientState.hs

```diff
@@ -84,7 +82,7 @@ data FetchClientPolicy header block m =
        blockFetchSize     :: header -> SizeInBytes,
        blockMatchesHeader :: header -> block -> Bool,
        addFetchedBlock    :: Point block -> block -> m (),
-       headerForgeUTCTime :: FromConsensus header -> STM m UTCTime
+       headerForgeUTCTime :: header -> UTCTime
```

Comment:
This new type is a much stronger requirement on the `header` type, but the Consensus Layer already supports it. So that seems derisked, from my perspective.